### PR TITLE
ci: update docker image for ci

### PR DESCRIPTION
**Pull request by wep21**
update docker image from `focal` to  `jammy`

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -3,12 +3,14 @@ name: Lint
 on:
   pull_request:
 
+env:
+  AMENT_CPPCHECK_ALLOW_SLOW_VERSIONS: 1
```
Comment (christophebedard) [inline .github/workflows/lint.yml:7]: just wondering: why is this needed?

Comment (wep21) [reply .github/workflows/lint.yml:7]: This is a workaround for the error below. https://github.com/ros-tooling/topic_tools/actions/runs/3209171467/jobs/5245668834
```
cppcheck 2.7 has known performance issues and therefore will not be used, set the AMENT_CPPCHECK_ALLOW_SLOW_VERSIONS environment variable to override this.
```

Comment (christophebedard) [reply .github/workflows/lint.yml:7]: oh okay, thanks for the explanation!

**Review by christophebedard — APPROVED** [review]
